java/src/main/java/ai/rapids/cudf/Rmm.java

```diff
@@ -136,6 +136,52 @@ public static boolean isInitialized() throws RmmException {
    */
   public static native long getTotalBytesAllocated();
 
+  /**
+   * Returns the maximum amount of RMM memory (Bytes) outstanding during the
+   * lifetime of the process.
+   *
+   * Note that this result is meaningful when a single thread is using the GPU, or
+   * when we have joined all threads and CUDA synchronized with all streams.
+   */
+  public static native long getMaximumOutstanding();
```

Comment:
Would prefer this is called `getMaximumTotalBytesAllocated` for consistency with the above method, as the name shows it's clearly tied to that metric.